### DISCUSSION TITLE
Way to allocate serial numbers to sales order from list?

### DISCUSSION THREAD

**Discussion by runswithcode** [inventree/InvenTree#5877] [Q&A]
In most of InvenTree there's nice lists of items with filtering options and links and multiple selection options, but for allocating serial numbers to a sales order it's a manual blind text entry system that requires a second instance of InvenTree open to utilize efficiently. For most users with SN's these values are usually long and complex, so in digging around I'm still incredulous that I can't find and alternative system for this part of the process. I'll move this to a feature request if it doesn't exist, but I'm hoping I'm just being dumb and not finding the right way to do this! Thanks InvenTree gurus :)

**Comment by SchrodingersGat** [comment]
@runswithcode currently there are two distinct ways to allocate individual items to a sales order:

1. Select a single stock item with a drop-down filterable search input - here you can input the serial number you are looking for
2. Enter a set of text serial numbers, including range values, e.g. `100-105`

However you cannot combine the two.

The [new user interface based on react](https://github.com/inventree/InvenTree/issues/5212) would be better prepared to support some better user interface workflows such as this.

If you have any specific ideas around this please raise an issue for it

**Reply by runswithcode** [reply]
@SchrodingersGat thanks for the response! Yes number 1 is the solution i'm looking for. You say you can't combine both, but how would I switch to the one I want? I'm specifically thinking of the "Add Line Items" pop-up window in the "Sales Order" page. Is there a setting or change of html to do that?

I'll also look into the react UI, thanks.

**Reply by SchrodingersGat** [reply]
> You say you can't combine both, but how would I switch to the one I want?

Select the "allocate items" button shown here:

![image](https://github.com/inventree/InvenTree/assets/10080325/07b45403-3a69-4292-8e4a-e4b593853933)

**Reply by runswithcode** [reply]
Oh, I see! I never tried this buttons since I assumed I wanted to "Allocate Serial Numbers", but "Allocate Stock" is a better way to search for serial numbers. Thanks!
